File: solutions/csharp/binary-search/1/BinarySearch.cs
public static class BinarySearch
{
    public static int Find(int[] input, int value)
    {
        Array.Sort(input);
        int left = 0, right = input.Length - 1;
        while (left <= right) {
            int middle = (left + right) / 2;

            if (input[middle] == value) return middle;

            if (input[middle] < value) left = middle + 1;
            else right = middle - 1;
        }

        return -1;
    }
}

File: solutions/python/binary-search/1/binary_search.py
def find(search_list, value):
    search_list.sort()
    left = 0
    right = len(search_list) - 1
    while left <= right:
        middle = (left + right) // 2
        if search_list[middle] < value:
            left = middle + 1
        elif search_list[middle] > value:
            right = middle - 1
        else:
            return middle

    raise ValueError("value not in array")